test/unit/server/handlers/consents/{ID}/revoke.test.ts

```diff
@@ -0,0 +1,261 @@
+  })
+
+  describe('validateRequestAndRevokeConsent', (): void => {
+    it('Should finish with no errors',
+      async (): Promise<void> => {
+        await expect(Handler.validateRequestAndRevokeConsent(request))
+          .resolves.toBe(undefined)
+
+        expect(mockConsentRetrieve).toBeCalledWith(consentId)
+        expect(mockIsConsentRequestValid)
+          .toBeCalledWith(partialConsentActive, request)
+        expect(mockRevokeConsentStatus).toBeCalledWith(partialConsentActive)
+        expect(Domain.generatePatchConsentRequest)
+          .toBeCalledWith(partialConsentRevoked)
+        expect(mockPatchConsents)
+          .toBeCalledWith(consentId,
+            requestBody,
+            request.headers[Enum.Http.Headers.FSPIOP.SOURCE]
+          )
+      })
+
+    it('Should also finish with no errors',
+      async (): Promise<void> => {
+        mockConsentRetrieve.mockResolvedValueOnce(completeConsentRevoked)
+        mockRevokeConsentStatus.mockResolvedValueOnce(completeConsentRevoked)
+
+        await expect(Handler.validateRequestAndRevokeConsent(request))
+          .resolves.toBe(undefined)
+
+        expect(mockConsentRetrieve).toBeCalledWith(consentId)
+        expect(mockIsConsentRequestValid)
+          .toBeCalledWith(completeConsentRevoked, request)
+        expect(mockRevokeConsentStatus).toBeCalledWith(completeConsentRevoked)
+        expect(mockGeneratePatchConsentRequest)
+          .toBeCalledWith(completeConsentRevoked)
+        expect(mockPatchConsents)
+          .toBeCalledWith(
+            consentId,
+            requestBody,
+            request.headers[Enum.Http.Headers.FSPIOP.SOURCE]
+          )
+      })
+
+    it('Should throw an error due to consent retrieval error',
+      async (): Promise<void> => {
+        mockConsentRetrieve.mockRejectedValueOnce(new Error('Test'))
+        await expect(Handler.validateRequestAndRevokeConsent(request))
+          .rejects.toThrowError('NotImplementedYetError')
+
+        expect(mockConsentRetrieve).toBeCalledWith(consentId)
+        expect(mockIsConsentRequestValid).not.toBeCalled()
+        expect(mockRevokeConsentStatus).not.toBeCalled()
+        expect(mockGeneratePatchConsentRequest).not.toBeCalled()
+        expect(mockPatchConsents).not.toBeCalled()
+      })
+
+    it('Should throw an error due to invalid request',
+      async (): Promise<void> => {
+        mockIsConsentRequestValid.mockReturnValueOnce(false)
+        await expect(Handler.validateRequestAndRevokeConsent(request))
+          .rejects.toThrowError('NotImplementedYetError')
+
+        expect(mockConsentRetrieve).toBeCalledWith(consentId)
+        expect(mockIsConsentRequestValid).toBeCalledWith(partialConsentActive, request)
+        expect(mockRevokeConsentStatus).not.toBeCalled()
+        expect(mockGeneratePatchConsentRequest).not.toBeCalled()
+        expect(mockPatchConsents).not.toBeCalled()
+      })
+
+    it('Should throw an error as patchConsents() throws an error',
```

Comment:
```suggestion
    it('Should propagate errors from patchConsents()',
```
there is a need to improve the readability of `it` descriptions here and around